src/authentication/form/FlowType.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Controller, useFormContext } from "react-hook-form";
+import {
+  FormGroup,
+  Select,
+  SelectOption,
+  SelectVariant,
+} from "@patternfly/react-core";
+
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+
+const types = ["basic-flow", "client-flow"];
+
+export const FlowType = () => {
+  const { t } = useTranslation("authentication");
+  const { control } = useFormContext();
+
+  const [open, setOpen] = useState(false);
+
+  return (
+    <FormGroup
+      label={t("flowType")}
+      labelIcon={
+        <HelpItem
+          helpText="authentication-help:flowType"
+          forLabel={t("flowType")}
+          forID="flowType"
+        />
+      }
+      fieldId="flowType"
+    >
+      <Controller
+        name="providerId"
+        defaultValue={types[0]}
+        control={control}
+        render={({ onChange, value }) => (
+          <Select
+            toggleId="flowType"
+            onToggle={() => setOpen(!open)}
+            onSelect={(_, value) => {
+              onChange(value as string);
```

Comment:
A `SelectOptionObject` need to implement `toString()` so we do not have to type-cast this value.
```suggestion
              onChange(value.toString());
```